echo-pubsub-aws/src/main/java/com/netflix/spinnaker/echo/config/AmazonPubsubProperties.java

```diff
@@ -64,12 +69,19 @@ public AmazonPubsubSubscription(
       String topicARN,
       String queueARN,
       String templatePath,
-      MessageFormat messageFormat) {
+      MessageFormat messageFormat,
+      boolean idInMessageAttributes,
+      Long dedupeRetentionMillis
+      ) {
       this.name = name;
       this.topicARN = topicARN;
       this.queueARN = queueARN;
       this.templatePath = templatePath;
       this.messageFormat = messageFormat;
+      this.idInMessageAttributes = idInMessageAttributes;
+      if (dedupeRetentionMillis != null && dedupeRetentionMillis >= 0) {
```

Comment:
I think if you're doing constraint checking, you should throw an exception, or at minimum log that the configuration is invalid, rather than silently dropping invalid config.

Reply:
Okie